src/textual/app.py

```diff
@@ -1164,9 +1179,16 @@ def _start_widget(self, parent: Widget, widget: Widget) -> None:
         """
         widget._attach(parent)
         widget._start_messages()
-        widget.post_message_no_wait(events.Mount(sender=parent))
 
     def is_mounted(self, widget: Widget) -> bool:
```

Comment:
The difference between register and mount is confusing me - is registering simply "adding the widget to the registry", and mounting is "adding the widget to the DOM"?

Should this be `is_registered`?

Reply:
Pretty much. "register" is the internal name. "Mounting" is the full procedure which includes registering and sending events.

Should still be `is_mounted` since if it is in the registry the mount procedure will be complete.